src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,217 @@
+     * @param[in]  metadataOptions  Reserved for future use (value = 0).
+     * @param[in]  peerNodeId       The Node ID of the controller or requestor.
+     * @param[in]  peerFabricIndex  The FabricIndex of the requestor.
+     * @param[out] outSession       On success, must be populated with the new session's details.
+     * @param[out] outDeferredOffer True if the device is in standby or needs to defer generating an Offer.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR if a session was successfully created or prepared for negotiation
+     *   - Appropriate error otherwise (e.g. out of resources)
+     */
+    virtual CHIP_ERROR HandleSolicitOffer(uint16_t id, StreamUsageEnum streamUsage,
+                                          const Optional<DataModel::Nullable<uint16_t>> & videoStreamId,
+                                          const Optional<DataModel::Nullable<uint16_t>> & audioStreamId,
+                                          const Optional<DataModel::DecodableList<ICEServerStruct>> & iceServers,
+                                          const Optional<chip::CharSpan> & iceTransportPolicy,
+                                          const Optional<chip::BitMask<WebRTCMetadataOptionsBitmap>> & metadataOptions,
+                                          NodeId peerNodeId, FabricIndex peerFabricIndex, WebRTCSessionStruct & outSession,
+                                          bool & outDeferredOffer) = 0;
+
+    /**
+     * @brief
+     *   This method is called when the server receives the ProvideOffer command.
+     *
+     *   ProvideOffer is used by a requestor to either:
+     *     - Create a brand new session by sending an SDP Offer
+     *     - Re-offer an SDP to modify an existing session (e.g., enabling two-way talk).
+     *
+     * @param[in]  id                The existing session ID or 0 if new (depending on your usage).
+     * @param[in]  streamUsage       The stream usage (camera feed, doorbell, etc.).
+     * @param[in]  sdp               The SDP Offer (RFC 8866).
+     * @param[in]  videoStreamId     Requested or existing VideoStreamID.
+     * @param[in]  audioStreamId     Requested or existing AudioStreamID.
+     * @param[in]  iceServers        Optional ICE server list.
+     * @param[in]  iceTransportPolicy If present, indicates 'all' or 'relay' for candidate gathering.
+     * @param[in]  metadataOptions   Reserved (value = 0).
+     * @param[in]  peerNodeId        Requestor’s NodeId.
+     * @param[in]  peerFabricIndex   Requestor’s FabricIndex.
+     * @param[out] outSession        Must be populated with the final session info (session ID, streams, etc.).
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR on success
+     *   - Appropriate error if the request is invalid or resources are exhausted
+     */
+    virtual CHIP_ERROR HandleProvideOffer(uint16_t id, StreamUsageEnum streamUsage, const chip::CharSpan & sdp,
```

Comment:
same here: too many arguments. lets not have methods with more than say 4 arguments.